internal/watcher/instance/instance_watcher_service.go

```diff
@@ -248,14 +250,26 @@ func checkForProcessChanges(
 ) (updatedInstances []*mpi.Instance) {
 	for instanceID, instance := range updatedInstancesMap {
 		oldInstance := updatedOldInstancesMap[instanceID]
-		if oldInstance.GetInstanceRuntime().GetProcessId() != instance.GetInstanceRuntime().GetProcessId() {
+		if compareInstance(oldInstance.GetInstanceRuntime(), instance.GetInstanceRuntime()) {
 			updatedInstances = append(updatedInstances, instance)
 		}
 	}
 
 	return updatedInstances
 }
 
+func compareInstance(oldRuntime, currentRuntime *mpi.InstanceRuntime) (updated bool) {
```

Comment:
nit: `compareInstance` sounds like it's used to compare order of instances in a list. I'd consider renaming this to `areInstancesEqual` or `instancesEqual` or even the Java-style `isEqual`.

You'd have to invert the booleans in the function and add a `!` to the if clause, but I think it would slightly improve readability. Not a big deal if you wanna leave it as is though.

Reply:
Done